### PR TITLE
in_prometheus_remote_write: Add documentation for input plugin of prometheus remote write

### PR DESCRIPTION
**Pull request by cosmo0920**
Corresponding to https://github.com/fluent/fluent-bit/pull/8725.

### CODE REVIEW

File: pipeline/inputs/prometheus-remote-write.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+description: An input plugin to ingest payloads of Prometheus remote write
+---
+
+# Prometheus Remote Write
+
+The input plugin of Prometheus remote write allows you to ingest payloads of Prometheus remote write protocol.
+
+The OpenTelemetry plugin allows you to ingest telemetry data as per the OTLP specification, from various OpenTelemetry exporters, the OpenTelemetry Collector, or Fluent Bit's OpenTelemetry output plugin.
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md:9]: My bad.... I'll fix it.

File: pipeline/inputs/prometheus-remote-write.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+| ----------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | ------- |
+| listen            | The address to listen on                                                                                                                       | 0.0.0.0 |
+| port              | The port for Fluent Bit to listen on                                                                                                           | 80    |
+| buffer\_max\_size   | Specify the maximum buffer size in KB to receive a JSON message.                                                                               | 4M      |
+| buffer\_chunk\_size | This sets the chunk size for incoming incoming JSON messages. These chunks are then stored/managed in the space available by buffer_max_size.  | 512K    |
+|successful\_response\_code | It allows to set successful response code. `200`, `201` and `204` are supported.| 201 |
+| tag\_from\_uri      | If true, tag will be created from uri. e.g. v1_metrics from /v1/metrics .                                                                      | true    |
+
+
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md:22]: Yes. We are always able to use TLS. I'll check it.

File: pipeline/inputs/prometheus-remote-write.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+The input plugin of Prometheus remote write allows you to ingest payloads of Prometheus remote write protocol.
+
+The OpenTelemetry plugin allows you to ingest telemetry data as per the OTLP specification, from various OpenTelemetry exporters, the OpenTelemetry Collector, or Fluent Bit's OpenTelemetry output plugin.
+
+## Configuration <a href="#configuration" id="configuration"></a>
+
+| Key           | Description                                                                                                                                    | default |
+| ----------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | ------- |
+| listen            | The address to listen on                                                                                                                       | 0.0.0.0 |
+| port              | The port for Fluent Bit to listen on                                                                                                           | 80    |
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md:16]: Hmm.., maybe 8080 should be default port for HTTP listen?

File: pipeline/inputs/prometheus-remote-write.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+| buffer\_max\_size   | Specify the maximum buffer size in KB to receive a JSON message.                                                                               | 4M      |
+| buffer\_chunk\_size | This sets the chunk size for incoming incoming JSON messages. These chunks are then stored/managed in the space available by buffer_max_size.  | 512K    |
+|successful\_response\_code | It allows to set successful response code. `200`, `201` and `204` are supported.| 201 |
+| tag\_from\_uri      | If true, tag will be created from uri. e.g. v1_metrics from /v1/metrics .                                                                      | true    |
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md]: tag_from_uri is always prioritized to create tag for ingested metrics events and this should be true be default.
Without tag_from_uri should require to specify tag parameter explicitly.

File: pipeline/inputs/prometheus-remote-write.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md:20]: I confirmed that we can use TLS but it also depends on Prometheus' configuration for TLS.
We didn't ensure other products configurations. Should we add an example for communicating with TLS between fluent-bit and Prometheus server?

[the remark this answers is not in the document]
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md:20]: I added an example for TLS: https://github.com/fluent/fluent-bit-docs/pull/1363/commits/684be67275999e6c220fa91269dbc3d33d1d4bd2

File: pipeline/inputs/prometheus-remote-write.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md]: I added additional descriptions for tag and tag_from_uri parameters: https://github.com/fluent/fluent-bit-docs/pull/1363/commits/aced09ecf09dad3ffa9724ee1b37f8cd6a3083a1

File: pipeline/inputs/prometheus-remote-write.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+| listen            | The address to listen on                                                                                                                       | 0.0.0.0 |
+| port              | The port for Fluent Bit to listen on                                                                                                           | 8080    |
+| buffer\_max\_size   | Specify the maximum buffer size in KB to receive a JSON message.                                                                               | 4M      |
+| buffer\_chunk\_size | This sets the chunk size for incoming incoming JSON messages. These chunks are then stored/managed in the space available by buffer_max_size.  | 512K    |
+|successful\_response\_code | It allows to set successful response code. `200`, `201` and `204` are supported.| 201 |
+| tag\_from\_uri      | If true, tag will be created from uri. e.g. api\_prom\_push from /api/prom/push. This parameter will be prioritized when tag parameter is specified. If specify as false, it needs to specify user specified tag with tag parameter. | true    |
```
Comment (patrick-stephens) [inline pipeline/inputs/prometheus-remote-write.md:18]:
```suggestion
| tag\_from\_uri      | If true, tag will be created from uri, e.g. api\_prom\_push from /api/prom/push, and any tag specified in the config will be ignored. If false then a tag must be provided in the config for this input. | true    |
```

File: pipeline/inputs/prometheus-remote-write.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+{% endtab %}
+{% endtabs %}
+
+With the above configuration, Fluent Bit will listen on port `8080` for data. You can now send payloads of Prometheus remote write to the endpoints `/api/prom/push`.
+
+## Examples
+
+### Communicate with TLS
+
+Communicating with TLS, you will need to use the tls related parameters:
```
Comment (patrick-stephens) [inline pipeline/inputs/prometheus-remote-write.md:61]: I'd link out to the transport-security doc as well.

File: pipeline/inputs/prometheus-remote-write.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+	Name prometheus_remote_write
+	Listen 127.0.0.1
+	Port 8080
+	Uri /api/prom/push
+	Tls On
+	tls.crt_file /path/to/certificate.crt
+	tls.key_file /path/to/certificate.key
+```
+
+Then, enabled TLS configuration and able to ingest payloads of metrics with Prometheus' remote write protocol.
```
Comment (patrick-stephens) [inline pipeline/inputs/prometheus-remote-write.md:74]:
```suggestion
Now, you should be able to send data over TLS to the remote write input.
```

File: pipeline/inputs/prometheus-remote-write.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+          port: 8080
+          uri: /api/prom/push
+    outputs:
+        - name: stdout
+          match: '*'
+```
+{% endtab %}
+{% endtabs %}
+
+With the above configuration, Fluent Bit will listen on port `8080` for data. You can now send payloads of Prometheus remote write to the endpoints `/api/prom/push`.
```
Comment (patrick-stephens) [inline pipeline/inputs/prometheus-remote-write.md:55]:
```suggestion
With the above configuration, Fluent Bit will listen on port `8080` for data.
You can now send payloads in Prometheus remote write format to the endpoint `/api/prom/push`.
```

File: pipeline/inputs/prometheus-remote-write.md (hunk 6 above)
Comment (cosmo0920) [reply pipeline/inputs/prometheus-remote-write.md:63]: I added a link for it and added mention this plugin there: https://github.com/fluent/fluent-bit-docs/pull/1363/commits/0c1b1999c2caa63adf0303933bf912a1bc1f9c4a

**Comment by agup006** [comment]
As this was released in 3.0.4 will merge this

**Pull request merged**